src/redux/remoteData/asyncActionCreator.ts

```diff
@@ -15,39 +27,107 @@ export const getIssuesIfNeeded = () => {
         // tslint:disable-next-line:no-console
       }).catch((error) => console.error(`fetchCallCount error: ${error.message}`, error));
   };
 };
 
-// TODO: integrate geolocation and catch errors
 export const fetchLocationByIP = () => {
-  return (dispatch, getState) => {
-    api.getLocationByIP()
-      .then((response) => {
-        const location = response.data.loc;
-        dispatch(getIssues(location));
+  return (dispatch: Dispatch<ApplicationState>,
+          getState: () => ApplicationState) => {
+    clearTimeout(timeout);
+    // check to see if state contains an address already,
+    // which means that fetchGeolocation() has been successful
+    const state: ApplicationState = getState();
+    if (!state.locationState.address) {
+      dispatch(setFetchingLocation(true));
+      dispatch(setInvalidAddress(true));
+      getLocationByIP()
+        .then((response: IpInfoData) => {
+          const location = response.loc;
+          dispatch(getApiData(location));
+          dispatch(setFetchingLocation(false));
+          // TODO: dispatch an error message
+          // tslint:disable-next-line:no-console
+        }).catch((error) => console.error(`fetchLocationByIP error: ${error.message}`, error));
+    }
+  };
+};
+
+export const fetchBrowserGeolocation = () => {
+  return (dispatch: Dispatch<ApplicationState>,
+          getState: () => ApplicationState) => {
+    // Sometimes, the user ignores the prompt or the browser does not
+    // provide a response when they do not permit browser location.
+    // After GEOLOCATION_TIMEOUT + 1 second, try IP-based location,
+    // but let browser-based continue. This timeout is cleared after
+    // either geolocation or ipinfo.io location succeeds.
+    timeout = setTimeout(geolocationTimeoutHandler(dispatch, getState), GEOLOCATION_TIMEOUT + 1000);
+    dispatch(setFetchingLocation(true));
+    dispatch(setInvalidAddress(true));
+    getBrowserGeolocation()
+      .then(location => {
+        if (location.latitude && location.longitude) {
+          const loc = `${location.latitude} ${location.longitude}`;
+          dispatch(setFetchingLocation(true));
+          dispatch(getApiData(loc));
+          dispatch(setFetchingLocation(false));
+          clearTimeout(timeout);
+        } else {
+          // console.log('fetchGeolocation() no location found. Clearing address');
+          dispatch(setFetchingLocation(false));
+          fetchLocationByIP();
+        }
+      })
+      .catch(e => {
         // tslint:disable-next-line:no-console
-      }).catch((error) => console.error(`fetchLocationByIP error: ${error.message}`, error));
+        console.error('Problem getting browser geolocation', e);
+        fetchLocationByIP();
+      });
   };
-};
\ No newline at end of file
+};
+
+const geolocationTimeoutHandler = (dispatch, getState) => {
+  // tslint:disable-next-line:no-console
+  console.debug('geolocationTimeoutHandler() called');
```

Comment:
This is triggered nearly immediately on page load for me, even if I bump the timeout to something absurdly long. I can allow location from this page but it looks like it falls back to ipinfo every time for me.

I'm not sure how best to give you more information on this. Happy to have a pair debugging session if it's not obvious what's going on.